src/System.Threading.Thread/tests/ThreadTests.cs

```diff
@@ -506,12 +506,34 @@ public static void ManagedThreadIdTest()
         [Fact]
         public static void NameTest()
         {
-            var t = new Thread(() => { });
+            string name = "a";
+            Action waitForThread;
+            var t =
+                ThreadTestHelpers.CreateGuardedThread(out waitForThread, () =>
+                {
+                    var ct = Thread.CurrentThread;
+                    Assert.Equal(name, ct.Name);
+                    Assert.Throws<InvalidOperationException>(() => ct.Name = name + "b");
+                    Assert.Equal(name, ct.Name);
+                });
+            t.IsBackground = true;
             Assert.Null(t.Name);
-            t.Name = "a";
-            Assert.Equal("a", t.Name);
-            Assert.Throws<InvalidOperationException>(() => t.Name = "b");
-            Assert.Equal("a", t.Name);
+            t.Name = name;
```

Comment:
What happens if you explicitly set it to null?  Are you able to do so any number of times?

Reply:
Yes but only if the name was not set yet, I'll add some more asserts